## Generate MNP-Flex classifier report via API

The `main.sh` merges BAM files, calls modifications with modkit, prepares the input for MNP-Flex, and generates report via MNP-Flex API.

This fork is designed to generate input files for the MNP-Flex classifier based on the rapid-CNS2 Nextflow pipeline. It has been adapted specifically for Nanopore sequencing data that includes 5hmC and 5mC modifications (5mC_5hmC or 5mCG_5hmCG), whereas the original pipeline was designed primarily for 5mC-only data (5mC or 5mGC). 

MNP-Flex (https://mnp-flex.org) is a platform-independent methylation classification system, including an MGMT promoter methylation prediction model, designed for sequencing-based methylation data. It utilizes the latest CNS tumor methylation classification model from molecularneuropathology.org.

### Input

1. bam_dir, a directory containing basecalled and aligned (hg38) BAM files, with either 5mC, 5mC_5hmC, 5mGC, and 5mGC_5hmGC modifications.
2. Path to GRCh38.p14.genome.fa
3. Sample id

### Output

1. MNPFlex.subset.bed file, formatted as input of the MNP-Flex classifier.
2. MNPFlex.subset.pdf file, MNP-Flex report.
3. MNPFlex.subset.txt file, extracted classification scores and MGMT status.

## Overview from Rapid-CNS<sup>2</sup> 

The Rapid-CNS<sup>2</sup> nextflow pipeline is a bioinformatics workflow designed for comprehensive analysis of genomic and epigenomic data generated using adaptive sampling based sequencing of central nervous system (CNS) tumours. It performs tasks such as basecalling, variant calling, methylation analysis, structural variant calling, copy number variation calling, and provides a comprehensive molecular diagnostic-ready report.

This pipeline is implemented using Nextflow, allowing for easy execution and scalability on various compute environments, including local machines, clusters, and cloud platforms.

## Acknowledgements
We are extremely grateful to all our lab members and collaborators for their support! 
Keeping up with AI to make our life easier and to compensate for our (Areeba's) art skills, our logo was generated by DALL-E.

## Citation
If you use this pipeline, please cite our preprint:

Felix Sahm, Areeba Patel, Kirsten GÃ¶bel et al. Versatile, accessible cross-platform molecular profiling of central nervous system tumors: web-based, prospective multi-center validation, 10 April 2024, PREPRINT (Version 1) available at Research Square [https://doi.org/10.21203/rs.3.rs-4182910/v1]

## Contributions
Contributions are welcome! If you encounter any issues, have suggestions for improvements, or would like to contribute new features, please open an issue or pull request on this repository.

## License

This project is licensed under the Apache 2.0 license
